File: String/reverseWords.java
/**
    557. Reverse Words in a String 3
    
    Description:

    Given a string, you need to reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.

*/

public class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" ");
        StringBuilder sentence = new StringBuilder();

        for (int i = 0; i < words.length; i++)
        {
            StringBuilder word = new StringBuilder();
            word.append(words[i]);
            words[i] = word.reverse().toString();
            sentence.append(words[i]);
            sentence.append(" ");
        }
        return sentence.toString().trim();
    }
}

File: String/reverseWords.js
/**

  Description:

  Given a string, you need to reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.

*/

/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let arr = [];
    let word = '';
    for (let i = s.length -1; i >= 0; i--)
    {
      if (s[i] === ' ')
      {
        // The unshift() method adds one or more elements to the beginning of an array and returns the new length of the array.
        arr.unshift(word);
        word = '';
        continue;
      }
      word += s[i];
    }
    arr.unshift(word);
    // The join() method joins all elements of an array into a string and returns this string.
    return arr.join(' ');
};
